language-design.md

```diff
@@ -17,6 +17,97 @@ For each quote, describe what it was about the quote that led you pick it.
 
 **Response**
 
+*“According to findings by researchers from Southern Illinois University, this 
+reaction isn’t just because you’re a n00b: they found that Perl, a major programming 
+language used by untold zillions of developers, is no more intuitive to novices than a 
+language with a randomly generated syntax. Why shouldn’t those "interfaces" be humanely 
+designed? … They created a "placebo language" called Randomo, whose syntax was randomly 
+generated, to use in trials alongside Quorum and Perl. Novice programmers were able to write 
+sample programs more accurately in Quorum versus Perl — an interesting, but not terribly 
+surprising, result. More surprising was how Perl compared to Randomo. To quote the paper: 
+'Perl users were unable to write programs more accurately than those using a language designed 
+by chance.' … Every piece of software we use was written by other people, slaving away over 
+thousands of lines of code. Why shouldn’t those "interfaces" be as humanely designed as 
+the ones we tap and swipe?”* 
+
+This quote really resonated with me because my first experience with “programming” was with 
+MIT app inventor. Even as a middle schooler, I never had to look at documentation to 
+understand how to “write code” in app inventor. Each piece of code was represented in the 
+form of a “puzzle” piece:
+
+1. When one finished adding a piece of code to the chain of puzzle pieces, one could click 
+on the most recently added piece and it would display all the possible pieces that one could 
+append to the new chain of puzzle pieces.
+2. Representing the code as puzzle pieces allowed one to intuitively understand the syntax of 
+coding in app inventor -- it was obvious, from taking one look at the different puzzle pieces 
+available, to infer what pieces of code could be used in a succession, because certain types 
+of puzzle pieces could only fit with certain other types of puzzle pieces.
+
+However, I’m afraid that a language that is revised to involve “natural-language replacements 
+for some of the more abstruse syntax,” which can make writing and reading code more intuitive, 
+can also subject the coding language to being deemed only for novices, who need the syntactical 
+hand-holding to understand grammar that experienced coders have programmed in for years. 
+Unfortunately, introducing more “human-friendly” programming languages may subject a language and 
+its users to some of the biases that the Yang and Rabkin article mentioned, and further “set up 
+a hierarchy among programmers that systematically privileges certain groups” [Yang and Rabkin 2015]. 
+In the same way that “computer science PhDs explain they were embarrassed to know Python 
```

Comment:
Agreed!  It's so sad that individuals will judge others when they use an "easier" language.  In many cases, using an easier language simply means using tools better suited to the job.  How sad that using an efficient tool (which in many cases will be python because of its low startup costs) is seen as a negative and unfairly penalizes women and minorities!